registry/proxy/proxyblobstore.go

```diff
@@ -20,21 +20,15 @@ type proxyBlobStore struct {
 	localStore  distribution.BlobStore
 	remoteStore distribution.BlobService
 	scheduler   *scheduler.TTLExpirationScheduler
+	mu          sync.Mutex
+	inflight    map[digest.Digest]*inflightBlob
```

Comment:
It looks like you're using this map as a set (`inflightBlob` is an empty struct). As a slight optimization, I'd suggest using `map[digest.Digest]struct{}`. This will avoid extra garbage generated by pointers to empty structs. `map[digest.Digest]inflightBlob` would be equivalent, but `map[digest.Digest]struct{}` make it clear to readers that the map is being used as a set.